### PR TITLE
WIP: feat(): support ng add

### PR DESCRIPTION
**Pull request by mhartington**
#### Short description of what this resolves:
Add support for ngAdd 

Angular users looking to add Ionic to their project can simply run

```
ng add @ionic/angular
```

#### Changes proposed in this pull request:

- Add initial schematics for ionic/angular
- Add simple build step for schematics

**Ionic Version**: 4.x

### CODE REVIEW

File: angular/src/schematics/utils/devkit-utils/ast-utils.ts [hunk 1]
```diff
@@ -0,0 +1,571 @@
+/**
```
[the remark this answers is not in the document]
Comment (mhartington) [reply angular/src/schematics/utils/devkit-utils/ast-utils.ts]: Added

File: angular/src/schematics/add/index.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { Path, join, normalize } from '@angular-devkit/core';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { addPackageToPackageJson } from './../utils/package';
+import { addModuleImportToRootModule } from './../utils/ast';
+import { addStyle, getWorkspace } from './../utils/config';
+import { Schema as IonAddOptions } from './schema';
+
+function addIonicAngularToPackageJson(): Rule {
+  return (host: Tree) => {
+    addPackageToPackageJson(host, 'dependencies', '@ionic/angular', 'latest');
```
[the remark this answers is not in the document]
Comment (mhartington) [reply angular/src/schematics/add/index.ts]: I think it depends on the npm setup, but it should be fine

File: angular/tsconfig.schematics.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "allowUnreachableCode": false,
+    "lib": ["es2017", "dom"],
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "sourceMap": false,
+    "pretty": true,
+    "rootDir": "src/schematics",
+    "target": "es2015",
```
[the remark this answers is not in the document]
Comment (mhartington) [reply angular/tsconfig.schematics.json]: Done

File: angular/src/schematics/collection.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "../../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+  "ng-add": {
```
Comment (vitaliy-bobrov) [inline angular/src/schematics/collection.json]: Missed path to the `schema.json`:

"schema": "./add/schema.json",

Comment (mhartington) [reply angular/src/schematics/collection.json]: Thanks @vitaliy-bobrov 😄

Comment (vitaliy-bobrov) [reply angular/src/schematics/collection.json]: Glad to help! Just watched ngHouston with you ;)

**Pull request merged**